### PR TITLE
`auth login` options: `--browser <BROWSER>` and `--no-browser`

### LINKED ISSUE #62: Support not opening a browser window
The CLI should support something like `BROWSER=""` or `--no-browser` to override auto-opening (auth) pages in a browser. See discussion in oxidecomputer/cli#217.

### PR DESCRIPTION
**Pull request by ahl**
closes #62

### CODE REVIEW

File: cli/src/cmd_auth.rs [hunk 1]
```diff
@@ -124,20 +124,26 @@ pub fn parse_host(input: &str) -> Result<url::Url> {
...
+    with_token: bool,
 
     /// The host of the Oxide instance to authenticate with.
     /// This assumes http; specify an `http://` prefix if needed.
     #[clap(short = 'H', long, env = "OXIDE_HOST", value_parser = parse_host)]
-    pub host: url::Url,
+    host: url::Url,
+
+    /// Override the default browser when opening the authentication URL.
+    #[clap(long)]
```
Comment (karencfv) [inline cli/src/cmd_auth.rs]: User gets a nice error if they attempt to use both flags :)

```suggestion
    #[clap(long, conflicts_with = "no-browser")]
```

File: cli/src/cmd_auth.rs [hunk 2]
```diff
@@ -124,20 +124,26 @@ pub fn parse_host(input: &str) -> Result<url::Url> {
...
+    browser: Option<String>,
+
+    /// Print the authentication URL rather than opening a browser window.
+    #[clap(long = "no-browser")]
```
Comment (karencfv) [inline cli/src/cmd_auth.rs]:
```suggestion
    #[clap(long = "no-browser", conflicts_with = "browser")]
```

**Review by david-crespo — APPROVED** [review]
lgtm pending karen's additions

**Comment by david-crespo** [comment]
On second thought, it already appears to not allow both args in a group by default. 


```
~/oxide/oxide-sdk-and-cli $ cargo run -- auth login --host 'a' --browser abc --no-browser
    Finished dev [unoptimized + debuginfo] target(s) in 0.11s
     Running `target/debug/oxide auth login --host a --browser abc --no-browser`
error: the argument '--browser <BROWSER>' cannot be used with '--no-browser'

Usage: oxide auth login --host <HOST> --browser <BROWSER>
```

The [docs](https://docs.rs/clap/latest/clap/struct.ArgGroup.html) are very vague about the default behavior of a non-required group.

File: cli/src/cmd_auth.rs (hunk 1 above)
Comment (ahl) [reply cli/src/cmd_auth.rs]: it was already working that way, albeit in a weird way that I had copied from a `typify` community contribution. Made it more normal.

**Pull request merged**